Assets/Plugins/Clean/Editor/ComponentUpgrade.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using TMPro;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+using Object = UnityEngine.Object;
+
+namespace Plugins.Clean.Editor
+{
+    public static class ComponentUpgrade
+    {
+        public static TextMeshProUGUI UpgradeText(Text t)
+        {
+            // Copy properties
+            var go = t.gameObject;
+            var properties = new ComponentProperties.Text(t);
+
+            // Swap Text component for TextMeshPropUGUI
+            Object.DestroyImmediate(t, true);
+            var tmp = go.AddComponent<TextMeshProUGUI>();
+
+            // Paste new properties
+            properties.Apply(tmp);
+            return tmp;
+        }
+
+        public static TMP_InputField UpgradeInputField(InputField input)
+        {
+            // Copy properties
+            var go = input.gameObject;
+            var properties = new ComponentProperties.InputField(input);
+
+            // ComponentUpgrade child components
+            TextMeshProUGUI textComponent = UpgradeText(input.textComponent);
```

Comment:
Any reason why you're using var above and TextMeshProUGUI here? I personally am a fan of using var, it seems that ArtStorm is not using it too often.

Reply:
No reason, much prefer `var` in most places. Will update